pennylane/vqe/vqe.py

```diff
@@ -90,36 +92,41 @@ class Hamiltonian:
         Note that this issue also arises when calling the ``simplify()`` method.
     """
 
-    def __init__(self, coeffs, observables, simplify=False):
+    num_wires = qml.operation.AnyWires
+    num_params = 1
+    par_domain = "A"
+
+    def __init__(self, coeffs, observables, simplify=False, id=None):
 
-        if len(coeffs) != len(observables):
+        if qml.math.shape(coeffs)[0] != len(observables):
             raise ValueError(
                 "Could not create valid Hamiltonian; "
                 "number of coefficients and operators does not match."
             )
 
-        if any(np.imag(coeffs) != 0):
-            raise ValueError(
-                "Could not create valid Hamiltonian; " "coefficients are not real-valued."
-            )
```

Comment:
Do we need this? We don't check anything else here, and operator arithmetic could make sense with complex coefficients as long as the result of the cost function is real.

Reply:
💯 it is on our roadmap to remove this!